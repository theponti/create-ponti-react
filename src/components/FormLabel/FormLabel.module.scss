.container {
  letter-spacing: 1.25px;
  font-weight: 600;
}
