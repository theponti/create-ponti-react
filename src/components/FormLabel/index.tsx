import t, { InferProps } from 'prop-types';
import styles from './FormLabel.module.scss';

function FormLabel({ children, htmlFor }: InferProps<typeof FormLabel.propTypes>) {
  return <label className={styles.container} htmlFor={htmlFor}>{children}</label>;
}

FormLabel.propTypes = {
  children: t.node.isRequired,
  htmlFor: t.string.isRequired,
};

export default FormLabel;
