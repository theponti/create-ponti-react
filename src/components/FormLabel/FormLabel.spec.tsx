import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import FormLabel from '.';

describe('<FormLabel/>', () => {
  it('should render component', () => {
    const { container } = render(<FormLabel htmlFor="name">Name</FormLabel>);
    expect(container).toMatchSnapshot();
  });
});
