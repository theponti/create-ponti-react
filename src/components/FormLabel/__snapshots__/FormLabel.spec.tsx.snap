// Vitest Snapshot v1

exports[`<FormLabel/> > should render component 1`] = `
<div>
  <label
    class="_container_co5yv_1"
    for="name"
  >
    Name
  </label>
</div>
`;
